api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -356,6 +356,41 @@ const (
 	AWSMachinePublicIPs = "hypershift.openshift.io/aws-machine-public-ips"
 )
 
+// RetentionPolicy defines the policy for handling resources associated with a cluster when the cluster is deleted.
```

Comment:
just for organizational purposes I'd prefer to keep this self-contained within the Openstack file. Then me move it here if it's consumed agnostically or by more platforms

Reply:
fwiw I'd definitely like to adopt this for other platforms in future

Reply:
this is resolved but the declaration is still here, can we move it to openstack.go?